src_python/habitat_sim/utils/datasets_download.py

```diff
@@ -499,63 +509,66 @@ def clean_data(uid, data_path):
             os.rmdir(meta_dir)
 
 
-def download_and_place(
-    uid,
-    data_path,
+def clone_repo_source(
+    uid: str,
+    version_dir: str,
+    requires_auth: bool,
     username: Optional[str] = None,
     password: Optional[str] = None,
-    replace: Optional[bool] = None,
+    prune_lfs: bool = True,
 ):
-    r"""Data-source download function. Validates uid, handles existing data version, downloads data, unpacks, writes version, cleans up."""
-    if not data_sources.get(uid):
-        print(f"Data download failed, no datasource named {uid}")
-        return
-
-    # link_path = os.path.join(data_path, data_sources[uid]["link"])
-    link_path = pathlib.Path(data_sources[uid]["link"])
+    """
+    Clones and processes a datasource hosted on a git repo (e.g. HuggingFace Dataset).
+    Handles authentication for gated sources.
+    Automatically prunes the resulting repo to reduce memory overhead.
+    """
     version_tag = data_sources[uid]["version"]
-    version_dir = get_version_dir(uid, data_path)
-    downloaded_file_list = get_downloaded_file_list(uid, data_path)
+    clone_command = f" git clone --depth 1 --branch {version_tag} "
```

Comment:
Should we check whether `lfs` is installed here?

Reply:
lfs is installed as a system package because this is a commandline wrapper. I'm not sure exactly how to check if it is installed or that the version is correct from within the python process.

Reply:
We could check if `git lfs --version` returns something.

Reply:
Maybe something like:
```
if "git-lfs" not in subprocess.run(['git', 'lfs', 'version'], stdout=subprocess.PIPE).stdout.decode('utf-8'):
   <insert warning here>
```

Reply:
@0mdc I added checks in a few places, some assertions and some warnings. Also addressed pulling lfs assets from links if lfs was not installed on initial clone. Would be great if you could take another look and do some quick testing. :)

Reply:
Looks good!